hadoop-ozone/recon/src/test/java/org/apache/hadoop/ozone/recon/api/TestOmDBInsightEndPoint.java

```diff
@@ -0,0 +1,329 @@
+        .get("/sampleVol/bucketOne/key_one");
+    Assertions.assertEquals("key_one", omKeyInfo1.getKeyName());
+    RepeatedOmKeyInfo repeatedOmKeyInfo = new RepeatedOmKeyInfo(omKeyInfo);
+    reconOMMetadataManager.getDeletedTable()
+        .put("/sampleVol/bucketOne/key_one", repeatedOmKeyInfo);
+    RepeatedOmKeyInfo repeatedOmKeyInfo1 =
+        reconOMMetadataManager.getDeletedTable()
+            .get("/sampleVol/bucketOne/key_one");
+    Assertions.assertEquals("key_one",
+        repeatedOmKeyInfo1.getOmKeyInfoList().get(0).getKeyName());
+    Response deletedKeyInfo = omdbInsightEndpoint.getDeletedKeyInfo(-1, "");
+    KeyInsightInfoResp keyInsightInfoResp =
+        (KeyInsightInfoResp) deletedKeyInfo.getEntity();
+    Assertions.assertNotNull(keyInsightInfoResp);
+    Assertions.assertEquals("key_one",
+        keyInsightInfoResp.getRepeatedOmKeyInfoList().get(0).getOmKeyInfoList()
+            .get(0).getKeyName());
+  }
+
+  private OmKeyInfo getOmKeyInfo(String volumeName, String bucketName,
+                                 String keyName) {
+    return new OmKeyInfo.Builder()
+        .setVolumeName(volumeName)
+        .setBucketName(bucketName)
+        .setKeyName(keyName)
+        .setReplicationConfig(StandaloneReplicationConfig
+            .getInstance(HddsProtos.ReplicationFactor.ONE))
+        .setDataSize(random.nextLong())
+        .build();
+  }
+
+  @Test
+  public void testGetDeletedContainerKeysInfo() throws Exception {
+    Map<Long, ContainerMetadata> omContainers =
+        reconContainerMetadataManager.getContainers(-1, 0);
+    putContainerInfos(2);
+    List<ContainerInfo> scmContainers = reconContainerManager.getContainers();
+    assertEquals(omContainers.size(), scmContainers.size());
+    // Update container state of Container Id 1 to CLOSING to CLOSED
+    // and then to DELETED
+    reconContainerManager.updateContainerState(ContainerID.valueOf(1),
+        HddsProtos.LifeCycleEvent.FINALIZE);
+    reconContainerManager.updateContainerState(ContainerID.valueOf(1),
+        HddsProtos.LifeCycleEvent.CLOSE);
+    reconContainerManager
+        .updateContainerState(ContainerID.valueOf(1),
+            HddsProtos.LifeCycleEvent.DELETE);
+    Set<ContainerID> containerIDs = containerStateManager
+        .getContainerIDs(HddsProtos.LifeCycleState.DELETING);
+    Assert.assertEquals(1, containerIDs.size());
+
+    reconContainerManager
+        .updateContainerState(ContainerID.valueOf(1),
+            HddsProtos.LifeCycleEvent.CLEANUP);
+    containerIDs = containerStateManager
+        .getContainerIDs(HddsProtos.LifeCycleState.DELETED);
+    Assert.assertEquals(1, containerIDs.size());
+
+    List<ContainerInfo> deletedSCMContainers =
+        reconContainerManager.getContainers(HddsProtos.LifeCycleState.DELETED);
+    assertEquals(1, deletedSCMContainers.size());
+
+    Response deletedContainerKeysInfo =
+        omdbInsightEndpoint.getDeletedContainerKeysInfo(-1, "");
```

Comment:
Currently, the `testGetDeletedContainerKeysInfo()` method is only testing a single case. It would be good to test multiple cases such as when there are multiple open keys in the open key table or when the `limit` and `prevKeyPrefix` query parameters are used.

Reply:
@ArafatKhan2198  - thanks for review. I have added few more test cases.